src/VisualStudio/Core/Def/Implementation/AbstractEditorFactory.cs

```diff
@@ -331,11 +332,20 @@ private void FormatDocumentCreatedFromTemplate(IVsHierarchy hierarchy, uint item
             Contract.ThrowIfNull(rootToFormat);
             var documentOptions = ThreadHelper.JoinableTaskFactory.Run(() => addedDocument.GetOptionsAsync(cancellationToken));
 
+            // Add access modifier
+            var rootWithAccessModifier = ThreadHelper.JoinableTaskFactory.Run(() =>
```

Comment:
ick.  instead of this, can we move this all the way to the entrypoint and have it jsut call a clean async method?

Reply:
I'm not sure which entrypoint did you mean?

This is where the things are done currently:

https://github.com/dotnet/roslyn/blob/a358ab7e09565f99d5fb3c2521d782c1443df55b/src/VisualStudio/Core/Def/Implementation/AbstractEditorFactory.cs#L274-L285

Probably not very trivial to make it a clean async method.